File: Assets/Scripts/Core/FMonoModule.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class FMonoModule : MonoBehaviour, FModuleInterface
{
    public F_MODULE_STATE STATE
    {
        get;
        set;
    }

    public virtual int RelativeOrder => 100;

    public virtual IEnumerator OnPrepare()
    {
        STATE = F_MODULE_STATE.RUNNING;
        yield break;
    }

    public virtual void OnInitialize()
    {
      
    }

    public virtual void OnPause()
    {
       
    }


    public virtual void OnRefresh()
    {
        
    }

    public virtual void OnResume()
    {
      
    } 

    public virtual void Restart()
    {

    }
}

File: Assets/.lpproj/Apis/Unity/CS/FMonoModule.lua
---@class CS.FMonoModule : CS.UnityEngine.MonoBehaviour
CS.FMonoModule = {}

---@property readwrite CS.FMonoModule.STATE : CS.F_MODULE_STATE
CS.FMonoModule.STATE = nil

---@property readonly CS.FMonoModule.Name : CS.System.String
CS.FMonoModule.Name = nil

---@return CS.FMonoModule
function CS.FMonoModule()
end

---@return CS.System.Collections.IEnumerator
function CS.FMonoModule:OnPrepare()
end

function CS.FMonoModule:OnInitialize()
end

function CS.FMonoModule:OnPause()
end

function CS.FMonoModule:OnRefresh()
end

function CS.FMonoModule:OnResume()
end

function CS.FMonoModule:Restart()
end